relay-metrics/benches/benches.rs

```diff
@@ -1,135 +1,197 @@
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..1),
+                timestamp_shifts_generator: NumbersGenerator::new(1..10),
+            },
+        ),
+        (
+            "same metric on different projects",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..1),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(1..10),
+            },
+        ),
+        (
+            "all backdated metrics",
+            BucketsGenerator {
+                percentage_backdated: 1.0,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "all non-backdated metrics",
+            BucketsGenerator {
+                percentage_backdated: 0.0,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "a lot of metrics from a lot of projects with mix of backdated buckets and non",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 1_000_000,
+                metric_ids_generator: NumbersGenerator::new(1..1_000_000),
+                project_keys_generator: NumbersGenerator::new(1..1_000_000),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
     ];
 
-    for input in &inputs {
+    for (input_name, input) in &inputs {
         c.bench_with_input(
-            BenchmarkId::new("bench_insert_metrics", input),
+            BenchmarkId::new("bench_insert_metrics", input_name),
             &input,
-            |b, &input| {
+            |b, input| {
                 b.iter_batched(
                     || {
+                        let timestamp = UnixTimestamp::now();
                         let aggregator: Aggregator = Aggregator::new(config.clone());
-                        (aggregator, input.get_buckets())
+                        (aggregator, input.get_buckets(timestamp))
                     },
                     |(mut aggregator, buckets)| {
                         for (project_key, bucket) in buckets {
-                            aggregator.merge(project_key, bucket, None).unwrap();
+                            #[allow(clippy::unit_arg)]
+                            black_box(aggregator.merge(project_key, bucket, None).unwrap());
```

Comment:
I think the black box should be on the aggregator after the loop.

Reply:
I thought that `black_box` is handled specifically by the compiler just by wrapping an expression with it. Maybe I need to better read the docs.

Reply:
Okay, read the docs, you are right. It actually tries to be pessimistic about the dummy value.

Reply:
I read the docs, it's indeed doing some stuff with the result value, treating it pessimistically.

Reply:
The result value here is just the unit type. I am not sure what the compiler is allowed to optimize here, but for me it makes more sense on the "result".